erpnext/buying/doctype/purchase_order/purchase_order.js

```diff
@@ -94,7 +94,7 @@ erpnext.buying.PurchaseOrderController = erpnext.buying.BuyingController.extend(
 	get_items_from_open_material_requests: function() {
 		erpnext.utils.map_current_doc({
 			method: "erpnext.stock.doctype.material_request.material_request.make_purchase_order_based_on_supplier",
-			source_name: this.frm.doc.supplier,
+			source_doctype: "Supplier",
```

Comment:
Why this? It needs `source_name` which is selected supplier in the PO, not `source_doctype`.